tests/dialects/test_spark.py

```diff
@@ -214,6 +214,20 @@ def test_spark(self):
         self.validate_identity("TRIM(TRAILING 'SL' FROM 'SSparkSQLS')")
         self.validate_identity("SPLIT(str, pattern, lim)")
 
+        self.validate_all(
+            "BOOLEAN(x)",
+            write={
+                "": "CAST(x AS BOOLEAN)",
+                "spark": "CAST(x AS BOOLEAN)",
+            },
+        )
+        self.validate_all(
+            "STRING(x)",
+            write={
+                "": "CAST(x AS STRING)",
```

Comment:
The base generator ("SQLGlot" dialect) generates `TEXT` instead of `STRING`.
```suggestion
                "": "CAST(x AS TEXT)",
```